hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -110,26 +189,110 @@
     Collection<Trigger> getAllTriggersByTag(String category, String name) throws Exception;
 
     /**
-     * Used to generate an explicit Trigger from a Tokenized Trigger.  The dataIdMap replaces the tokens in the
-     * Conditions with actual dataIds.
+     * Orphan a member trigger.  The member trigger will no longer inherit group updates.  It will be allowed
+     * to be independently updated.  It does maintain its group reference and can again be tied to the
+     * group via a call to {@link #unorphanMemberTrigger(String, String, Map, Map)}.
+     * @param tenantId Tenant where trigger is stored
+     * @param memberId The member triggerId
+     * @return the member trigger
+     * @throws NotFoundException if trigger is not found
+     * @throws Exception
+     * @see {@link #unorphanMemberTrigger(String, String, Map, Map)} to again have the trigger be a full group member.
+     */
+    Trigger orphanMemberTrigger(String tenantId, String memberId) throws Exception;
+
+    /**
+     * Un-orphan a member trigger.  The member trigger is again synchronized with the group definition. As an orphan
+     * it may have been altered in various ways. So, as when spawning a new member trigger, the context and dataIdMap
+     * are specified.
+     * <p>
+     * This is basically a convenience method that first performs a {@link #removeTrigger(String, String)} and
+     * then an {@link #addMemberTrigger(String, String, String, String, Map, Map)}. But the member trigger must
+     * already exist for this call to succeed. The trigger will maintain the same group, id, and name.
+     * </p>
      * @param tenantId Tenant where trigger is stored
-     * @param triggerId Trigger to be copied
+     * @param memberId The member triggerId
+     * @param memberContext The member triggerContext. If null the context is inherited from the member trigger
      * @param dataIdMap Tokens to be replaced in the new trigger
-     * @return a copy of original trigger
-     * @throws Exception on any problem
+     * @return the member trigger
+     * @throws NotFoundException if trigger is not found
+     * @throws Exception
+     * @see {@link #orphanMemberTrigger(String, String)} for setting a member to be an orphan.
      */
-    Trigger copyTrigger(String tenantId, String triggerId, Map<String, String> dataIdMap) throws Exception;
+    Trigger unorphanMemberTrigger(String tenantId, String memberId, Map<String, String> memberContext,
```

Comment:
It is not possible to have an orphan trigger without a group trigger.  At group trigger removal time the orphan would have already been deleted or made a standard trigger. The same is true of the non-orphan members. See removeGroupTrigger() for more details.